src/Xamarin.Legacy.Sdk/Sdk/Xamarin.VsInstallRoot.targets

```diff
@@ -5,7 +5,9 @@
     <!-- Prefer ordering of 2022 Preview, 2022, 2019  -->
     <VsInstallRoot Condition=" '$(VsInstallRoot)' == '' and Exists ('$(ProgramFiles)/Microsoft Visual Studio/2022/Preview/') ">$(ProgramFiles)/Microsoft Visual Studio/2022/Preview/</VsInstallRoot>
     <VsInstallRoot Condition=" '$(VsInstallRoot)' == '' and Exists ('$(ProgramFiles)/Microsoft Visual Studio/2022/Enterprise/') ">$(ProgramFiles)/Microsoft Visual Studio/2022/Enterprise/</VsInstallRoot>
+    <VsInstallRoot Condition=" '$(VsInstallRoot)' == '' and Exists ('$(ProgramFiles)/Microsoft Visual Studio/2022/Community/') ">$(ProgramFiles)/Microsoft Visual Studio/2022/Community/</VsInstallRoot>
```

Comment:
Is there also a `Build Tools` package? Can we figure out what path it uses?

Reply:
There's a build tools package for VS2022 (and I suspect other versions of VS too) that one can install via:

```
winget install --id=Microsoft.VisualStudio.2022.BuildTools -e
```

On my machine that has VS2022 Community the aforementioned command created new folders under:

```
C:\Program Files\Microsoft Visual Studio\2022\Community
```

One of them being 'Common7' and under it the entire path:

```
C:\Program Files\Microsoft Visual Studio\2022\Community\Common7\IDE\ReferenceAssemblies\Microsoft\Framework
```

With its contents set to:

```
PS > ls 'C:\Program Files\Microsoft Visual Studio\2022\Community\Common7\IDE\ReferenceAssemblies\Microsoft\Framework' 


    Directory: C:\Program Files\Microsoft Visual Studio\2022\Community\Common7\IDE\ReferenceAssemblies\Microsoft\Framework


Mode                 LastWriteTime         Length Name
----                 -------------         ------ ----
d-----         28/9/2023   2:27 μμ                MonoAndroid
d-----         28/9/2023   2:29 μμ                MonoTouch
d-----         28/9/2023   2:29 μμ                Xamarin.iOS
d-----         28/9/2023   2:29 μμ                Xamarin.Mac
d-----         28/9/2023   2:29 μμ                Xamarin.TVOS
d-----         28/9/2023   2:29 μμ                Xamarin.WatchOS
```

I believe that this is what you're looking for.

Reply:
Yeah it's here: https://visualstudio.microsoft.com/downloads/

![image](https://github.com/xamarin/Xamarin.Legacy.Sdk/assets/840039/200bbacd-1363-4994-856b-085de315662c)

Reply:
Looks like it installs to `C:\Program Files (x86)\Microsoft Visual Studio\2022\BuildTools`, so maybe we add one more path here?

Reply:
> Yeah it's here: https://visualstudio.microsoft.com/downloads/
> 
> ![image](https://user-images.githubusercontent.com/840039/271319304-200bbacd-1363-4994-856b-085de315662c.png)

I downloaded and run the installer 'vs_BuildTools.exe' and it said 'already up-to-date' which probably means that the winget command essentially had the same effect.

> Looks like it installs to `C:\Program Files (x86)\Microsoft Visual Studio\2022\BuildTools`, so maybe we add one more path here?

True - it also seemed to install files there. However the following folder that &lt;TargetFrameworkRootPath&gt; aims for doesn't exist in my machine:

```
C:\Program Files (x86)\Microsoft Visual Studio\2022\BuildTools\Common7\IDE\ReferenceAssemblies\Microsoft\Framework
```

Maybe this folder was getting created for VS2019 which used to be 32bit but not anymore because VS2022 is 64bit from November 2021 onwards. So for VS2022 all build-tools folders seem to now be living under the following folder (as already mentioned above):

```
C:\Program Files\Microsoft Visual Studio\2022\Community\
```

which (quite naturally) is in the 64bit neighborhood of the Windows filesystem.

I also checked for the following folder but it didn't get created by the build-tools installer on machine:

```
C:\Program Files\Microsoft Visual Studio\2022\BuildTools    # not created
```

I believe I understand what you're after. I can add a heuristic specifically for VS2019 ala:

```
<VsInstallRoot Condition=" '$(VsInstallRoot)' == '' ">$(MSBuildProgramFiles32)/Microsoft Visual Studio/2019/BuildTools/</VsInstallRoot>
```

I think this folder makes sense for VS2019. Let me know if I'm missing something.

Reply:
> Wait, is `Professional` also not in the list?

Good catch! 'Professional' was also missing from the list. Adding it pronto!